et_replay/tools/comm_replay.py

```diff
@@ -832,10 +838,16 @@ def runComms(
             if self.is_blocking:
                 self.backendFuncs.complete_accel_ops(self.collectiveArgs)
 
-            # if nonblocking, then store the pair {reqID, future} so that we can wait on it later
+            # if nonblocking, then store the pair {(pg_id, reqID, isP2P), future} so that we can wait on it later
             # check if req id is recorded in trace for backwards compatibility
-            if curComm.req is not None and not self.is_blocking and collName != "wait":
-                self.collectiveArgs.waitObjIds[curComm.req] = retObj
+            if (
+                not self.is_blocking
+                and collName != "wait"
+                and self.collectiveArgs.wait_obj_key is not None
+            ):
+                self.collectiveArgs.waitObjIds[self.collectiveArgs.wait_obj_key] = (
+                    retObj
+                )
```

Comment:
Is parenthesis at 843 and 848 necessary?
![image](https://github.com/user-attachments/assets/75360794-85fc-453e-a33f-f7f7a7c0afa0)